packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,265 @@
+  @visibleForTesting
+  static final SensitiveContentHost instance = SensitiveContentHost._();
+
+  /// The state of content sensitivity in the widget tree.
+  ///
+  /// Contains the number of widgets with each [ContentSensitivity] level and
+  /// the current [ContentSensitivity] setting.
+  @visibleForTesting
+  ContentSensitivitySetting? getContentSenstivityState() {
+    return _contentSensitivitySetting;
+  }
+
+  /// Registers a [SensitiveContent] widget that will help determine the
+  /// [ContentSensitivity] level for the widget tree.
+  static Future<void> register(ContentSensitivity desiredSensitivityLevel) {
+    return instance._register(desiredSensitivityLevel);
+  }
+
+  Future<void> _register(ContentSensitivity desiredSensitivityLevel) async {
+    _contentSenstivityIsSupported ??= await _sensitiveContentService.isSupported();
+    if (!_contentSenstivityIsSupported!) {
+      // Setting content sensitivity is not supported on this device.
+      return;
+    }
+
+    // When the first `SensitiveContent` widget is registered, determine the content sensitivity
+    // level we should fallback to if/when no `SensitiveContent` widgets remain in the tree. This will be
+    // auto sensitive if it is otherwise unset by the developer. Also, initialize as the calculated
+    // content sensitivity level if we have not yet registered the first `SensitiveContent` widget
+    // to ensure we update that setting appropriately.
+    _fallbackContentSensitivitySetting ??= await _sensitiveContentService.getContentSensitivity();
+    calculatedContentSensitivityLevel ??= _fallbackContentSensitivitySetting;
+
+    // Update SensitiveContent widget count for those with desiredSensitivityLevel.
+    _contentSensitivitySetting.addWidgetWithContentSensitivity(desiredSensitivityLevel);
+
+    // Verify that desiredSensitivityLevel should be set in order for sensitive
+    // content to remain obscured.
+    if (calculatedContentSensitivityLevel ==
+        _contentSensitivitySetting.contentSensitivityBasedOnWidgetCounts) {
+      return;
+    }
+
+    // Set content sensitivity level as desiredSensitivityLevel. If the call to set content
+    // sensitivity on the platform side fails, then we do not update the current content
+    // sensitivity level.
+    _sensitiveContentService.setContentSensitivity(desiredSensitivityLevel);
+
+    // Update current content sensitivity level.
+    calculatedContentSensitivityLevel = desiredSensitivityLevel;
+  }
+
+  /// Unregisters a [SensitiveContent] widget from the [ContentSensitivitySetting] tracking
+  /// the content sensitivity level of the widget tree.
+  static void unregister(ContentSensitivity widgetSensitivityLevel) {
+    instance._unregister(widgetSensitivityLevel);
+  }
+
+  void _unregister(ContentSensitivity widgetSensitivityLevel) {
+    assert(
+      _contentSenstivityIsSupported != null,
+      'SensitiveContentHost.register must be called before SensitiveContentHost.unregister',
```

Comment:
Dont use asserts in java they are stripped out. Make an actual check and have an actual error or handle the condition and break out early.

Reply:
You mean in Dart? @justinmc recommended this assert for debug purposes: https://github.com/flutter/flutter/pull/158473#discussion_r1975956813

I think this is sufficient because this would only happen if someone is messing with the `SensitiveContent` widget internals.